### DISCUSSION TITLE
PDF support

### DISCUSSION THREAD

**Discussion by marcprux** [SusanDoggie/Doggie#9] [General]
I don't see many tests for PDFContext & PDFRenderer. Is it usable as a cross-platform replacement for CGContext's native PDF drawing?

**Comment by SusanDoggie** [comment]
Yes, it's cross-platform but many feature are not available, like text.
The whole thing is still in study.

The usage of PDFContext is same of the others Context like ImageContext and SVGContext.

PDFRenderer is the page renderer api allow you to render page into any context
You can find the entry in here https://github.com/SusanDoggie/Doggie/blob/c8186ed74cb44281ea28b69bdd20c7f3acfe916d/Sources/DoggieGraphics/PDFDocument/PDFRenderer/PDFDrawPage.swift#L28

this works like Apple’s CGContext.drawPDFPage
https://developer.apple.com/documentation/coregraphics/cgcontext/1456255-drawpdfpage

**Reply by SusanDoggie** [reply]
Technically, you can define your own DrawableContext to "read" the PDF draw commands. That’s how my app decode pdf file.

**Comment by SusanDoggie** [comment]
I add tests as examples
https://github.com/SusanDoggie/Doggie/blob/main/Tests/DoggieTests/PDFContextTest.swift
